src/net/metaplaylist/index.ts

```diff
@@ -0,0 +1,312 @@
+
+    const segmentPipeline = {
+      loader({
+        adaptation,
+        init,
+        manifest,
+        period,
+        representation,
+        segment,
+      } : ISegmentLoaderArguments) : ILoaderObservable<Uint8Array|ArrayBuffer> {
+        if (!segment.privateInfos) {
+          throw new Error("Segments from metaplaylist must have private infos.");
+        }
+        const transportType = getTypeFromPrivateInfos(segment.privateInfos);
+        const transport = transports[transportType];
+        const segmentLoader = transport.video.loader;
+        return segmentLoader({
+          adaptation,
+          init,
+          manifest,
+          period,
+          representation,
+          segment,
+        });
+      },
+
+      parser(args : ISegmentParserArguments<Uint8Array|ArrayBuffer>
+        ) : SegmentParserObservable {
+          const { period, init, segment } = args;
+          if (!segment.privateInfos) {
+            throw new Error("Segments from metaplaylist must have private infos.");
+          }
+          const transportType = getTypeFromPrivateInfos(segment.privateInfos);
+          const transport = transports[transportType];
+
+          if (!transport) {
+            throw new Error("Segments from metaplaylist must have private infos.");
+          }
+          const segmentParser = transport.video.parser;
+
+          return segmentParser(args).map(({ segmentData, segmentInfos }) => {
+            let segmentPatchedData;
+            const offset = (period.start || 0) *
+              (init ? (init.timescale || segment.timescale) :  segment.timescale);
+
+            if(segmentData !== undefined){
+              const responseData = segmentData instanceof Uint8Array ?
+                segmentData :
+                new Uint8Array(segmentData);
+
+              segmentPatchedData = new BoxPatcher(
+                responseData,
+                false,
+                false,
+                offset
+              ).filter();
+            }
+            if(segmentInfos){
+              segmentInfos.time += offset;
+            }
+            return { segmentData: segmentPatchedData, segmentInfos };
+          });
+
+        },
+      };
+
+    const textTrackPipeline = {
+      loader: (args: ISegmentLoaderArguments) => {
+        if (!args.segment.privateInfos) {
+          throw new Error("Segments from metaplaylist must have private infos.");
+        }
+        const transportType = getTypeFromPrivateInfos(args.segment.privateInfos);
+        const transport = transports[transportType];
+        return transport.text.loader(args);
+      },
+      parser: (args: ISegmentParserArguments<ArrayBuffer|string|Uint8Array>) => {
+        if (!args.segment.privateInfos) {
+          throw new Error();
+        }
+        const transportType = getTypeFromPrivateInfos(args.segment.privateInfos);
+        const transport = transports[transportType];
+        return transport.text.parser(args).map((_args) => {
```

Comment:
``_args`` is some fugly variable name! (arg is already ugly as fudge, we do not need to add another layer of sheet here)